esx_service/utils/auth_api.py

```diff
@@ -491,6 +491,10 @@ def vm_in_any_tenant(vms):
 def _tenant_vm_add(name, vm_list):
     """ API to add vms for a tenant """
     logging.debug("_tenant_vm_add: name=%s vm_list=%s", name, vm_list)
+    if  name  == auth_data_const.DEFAULT_TENANT:
```

Comment:
Suggestion: I think a good part of code for tenant vm operations is redundant across all the respective functions i.e.validating permission, getting tenant from db, validating vm_list (the common code). We can wrap it in a separate function (one or two) and call it in all required functions.